src/test/java/ObjectTest.java

```diff
@@ -905,8 +906,12 @@ public void testSSEKMSNotDeclared() {
 			AssertJUnit.fail("Expected Failure because of no x-amz-server-side-encryption header");
 		} catch (AmazonServiceException err) {
 			S3.logger.debug(String.format("TEST ERROR: %s%n", err.getMessage()));
-			AssertJUnit.assertEquals(err.getErrorMessage(),
-					"Server Side Encryption with KMS managed key requires HTTP header x-amz-server-side-encryption : aws:kms");
+			AssertJUnit.assertEquals("error code", "InvalidArgument",
+				err.getErrorCode());
+			AssertJUnit.assertEquals("status code", 400,
+				err.getStatusCode());
+			AssertJUnit.assertTrue("error message contains kms",
+				err.getErrorMessage().contains("kms"));
```

Comment:
yeah this change looks right to me. both the original error and your new one are 400 InvalidArgument, and both messages include 'kms'

is it possible that your teuthology run just didn't include this change? from [this job under rgw/verify](http://qa-proxy.ceph.com/teuthology/mwatts-2022-04-08_00:09:12-rgw-mdw-mankey-19-distro-default-smithi/6781806/teuthology.log) i see the override for:
```
    s3tests:
      force-branch: sse-s3-5
      git_remote: https://github.com/mdw-at-linuxbox/
```
but no override for `s3tests-java`, which uses these defaults:
```
  - s3tests-java:
      client.0:
        force-branch: ceph-master
        force-repo: https://github.com/ceph/java_s3tests.git
```

Reply:
@mdw-at-linuxbox was this comment addresses?